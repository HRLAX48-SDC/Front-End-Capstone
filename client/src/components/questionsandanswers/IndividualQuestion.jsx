import React, { useState, useEffect } from 'react';
import Answers from './Answers.jsx';
import { Button, Stack, Container, Row, Col } from 'react-bootstrap';
import axios from 'axios';
import API_KEY from '../../config/config.js';

const IndividualQuestion = (props) => {
  const question_id = props.question.question_id;

  const [q_helpful_count, setQCount] = useState(() => {
    return props.question.question_helpfulness;
  });
  const [markHelp, setMark] = useState(false);
  const markHelpful = () => {
    axios
      .put(`http://54.219.72.194/qa/questions/${question_id}/helpful`)
      .then((response) => {
        setQCount(q_helpful_count + 1);
        setMark(true);
      })
      .catch((err) => console.error(err));
  };

  const [reported, setReported] = useState(false);
  const markReport = () => {
    axios
      .put(`http://54.219.72.194/qa/questions/${question_id}/report`)
      .then((response) => {
        setReported(true);
      })
      .catch((err) => console.error(err));
  };

  return (
    <Container className='q_individual_q_grid' fluid='md'>
      <Row id='q_grid_row1'>
        <Col id='q_grid_row1_col1' xs={1}>
          <strong className='q_title'>Q:</strong>
        </Col>
        <Col id='q_grid_row1_col2'>
          <strong className='q_body'>{props.question.question_body}</strong>
        </Col>
        <Col id='q_grid_row1_col3' md={4}>
          <Stack direction='horizontal' gap={2}>
            <div className='ms-auto'>
              {!markHelp ? (
                <div>
                  <label>Helpful? </label>
                  <u id='q_helpful' onClick={() => markHelpful()}>
                    Yes
                  </u>
                  <span>({q_helpful_count})</span>
                </div>
              ) : (
                <div className='q_help_afterclick'>
                  <label>Helpful?</label>
                  <label>Yes({q_helpful_count})</label>
                </div>
              )}
            </div>
            <div className='vr' />
            <div>
              {!reported ? (
                <u id='q_report' onClick={() => markReport()}>
                  Report
                </u>
              ) : (
                <label className='q_reported'>Reported</label>
              )}
            </div>
          </Stack>
        </Col>
      </Row>
      <Answers question={props.question} product_name={props.product_name} />
      <br></br>
    </Container>
  );
};

export default IndividualQuestion;
