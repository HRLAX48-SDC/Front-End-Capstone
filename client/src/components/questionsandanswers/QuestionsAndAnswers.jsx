import React, { useState, useEffect} from 'react';
import axios from 'axios';
import {Button, Modal, Spinner} from 'react-bootstrap';
import API_KEY from '../../config/config.js';
import Questions from './Questions.jsx';


const QuestionsAndAnswers = (props) => {

  const [product_id, setId] = useState(props);
  useEffect(() => {
    setId(props.product.id)
  }, [props])

  //const [product_id, setId] = useState(42370);
  const [questions, setQuestions] = useState([]);

  useEffect(() => {
    getAllQuestions();
  }, [product_id]);

  const getAllQuestions = () => {
    axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-lax/qa/questions?product_id=${product_id}`, {
    headers: { 'Authorization': `${API_KEY}` }
    })
      .then((response) => {
        //console.log('questions',response.data.results);
        setQuestions(response.data.results);
      })
      .catch((err) => console.error(err));
  }

  const [numPerPage, setNumPerPage] = useState(4);
  const loadMore = () => {
    setNumPerPage(numPerPage + 2);
  }
  const sliceQuesions = questions.slice(0, numPerPage);
  return (
    <div className="questions-answers">
        <h5>QUESTIONS & ANSWERS</h5>
        <Questions questions={sliceQuesions} />
        <Button variant="outline-secondary" onClick={() => loadMore()}>LOAD MORE QUESTIONS</Button>
        <Button variant="outline-secondary">ADD A QUESTION +</Button>
    </div>
  );
};


export default QuestionsAndAnswers;
