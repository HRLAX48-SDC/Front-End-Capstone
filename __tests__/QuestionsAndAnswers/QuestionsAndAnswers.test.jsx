import React from 'react';
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import QuestionsAndAnswers from '../../client/src/components/questionsandanswers/QuestionsAndAnswers.jsx';
import IndividualQuestion from '../../client/src/components/questionsandanswers/IndividualQuestion.jsx';
Enzyme.configure({ adapter: new Adapter()});

const product = {
  id: 42366,
  campus: "hr-lax",
  name: "Camo Onesie",
  slogan: "Blend in to your crowd",
  updated_at: "2021-08-13T14:39:39.968Z",
  category: "Jackets",
  created_at: "2021-08-13T14:39:39.968Z",
  default_price: "140.00",
  description: "The So Fatigues will wake you up and fit you in. This high energy camo will have you blending in to even the wildest surroundings.",
 };
describe('Test Load More Questions Button component', () => {
  it('Test click event', () => {
    const mockCallBack = jest.fn();
    const wrapper = shallow(<QuestionsAndAnswers product={product}/>);
    wrapper.find('#loadBtn').simulate('click');
    //expect(wrapper.find(IndividualQuestion)).toHaveLength(6);
    expect(mockCallBack.mock.calls.length).toEqual(1);
  })
});