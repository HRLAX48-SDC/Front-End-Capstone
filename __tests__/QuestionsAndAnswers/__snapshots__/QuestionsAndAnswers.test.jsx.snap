// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load More Questions Button component Should render correctly 1`] = `null`;

exports[`Render Questions and Answers component correctly 1`] = `
<div
  className="questions-answers"
>
  <div
    className="questions_widget"
  >
    <div
      className="questions_list container"
    >
      <h5
        className="questions_widget_header"
      >
        QUESTIONS & ANSWERS
      </h5>
      <div
        className="row"
        id="q_list_search"
      >
        <form
          className=""
        >
          <input
            className="form-control"
            id="q_searchinput"
            onChange={[Function]}
            placeholder="ï€‚ HAVING A QUESTION? SEARCH FOR ANSWERS..."
            type="text"
            value=""
          />
        </form>
      </div>
      <div
        className="row"
        id="q_list_addQuestion"
      >
        <div
          id="add_question_button"
        >
          <label
            className="mb-3"
          >
            Don't see the question you're looking for?
          </label>
           
          <button
            className="question_add btn btn-outline-secondary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            ADD A QUESTION +
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
